src/numberAxis/Funnel.tsx

```diff
@@ -23,24 +23,25 @@ import {
   PresentationAttributesAdaptChildEvent,
   AnimationDuration,
 } from '../util/types';
+import { FunnelTrapezoid } from '../util/FunnelUtils';
 
-interface FunnelTrapezoidItem extends TrapezoidProps {
+export interface FunnelTrapezoidItem extends TrapezoidProps {
   value?: number | string;
   payload?: any;
+  isActive: boolean;
 }
 
 interface InternalFunnelProps {
-  trapezoids?: FunnelTrapezoidItem[];
-  animationId?: number;
-}
-
-interface FunnelProps extends InternalFunnelProps {
   className?: string;
   dataKey: DataKey<any>;
   nameKey?: DataKey<any>;
   data?: any[];
   hide?: boolean;
-  activeShape?: ReactElement<SVGElement> | ((props: any) => ReactElement<SVGElement>) | TrapezoidProps;
+  activeShape?:
+    | ReactElement<SVGProps<SVGPathElement>>
+    | ((props: FunnelTrapezoidItem) => ReactElement<SVGProps<SVGPathElement>>)
+    | SVGProps<SVGPathElement>
+    | boolean;
```

Comment:
We could extract a generic

```
type ActiveShape<T> = 
    | ReactElement<SVGProps<SVGPathElement>>
    | ((props: T) => ReactElement<SVGProps<SVGPathElement>>)
    | SVGProps<SVGPathElement>
    | boolean;
```

And reuse it in all places, right?

Reply:
So glad you suggested this! I started to do this previously but stopped. I've added this change and updated the other types to use it.